dockerfiles/vault-dcos/Readme.md

```diff
@@ -0,0 +1,114 @@
+# Vault on DC/OS
+
+This is an example of how to run [HashiCorp's Vault](https://github.com/hashicorp/vault) on DC/OS. By default, it's configured in HA mode with ZooKeeper as the backend.
+
+## Running on DC/OS
+
+### Step 1: Launch Vault server
+Save the following JSON as `vault.json`:
+```json
+{
+  "id": "vault",
+  "cpus": 1,
+  "mem": 1000,
+  "ports":[8200],
+  "requirePorts":true,
+  "instances": 1,
+  "container": {
+    "type": "DOCKER",
+    "docker": {
+      "image": "brndnmtthws/vault-dcos",
+      "network": "HOST",
+      "forcePullImage": true
+    }
+  },
+  "healthChecks": [{
+      "protocol": "TCP",
+      "portIndex": 0,
+      "timeoutSeconds": 10,
+      "gracePeriodSeconds": 10,
+      "intervalSeconds": 2,
+      "maxConsecutiveFailures": 10
+  }]
+}
+```
+Now launch on Marathon:
+```
+$ dcos marathon app add vault.json
+```
+
+**Note**: you may configure some parameters via environment variables, from the wrapper script [`run-vault`](run-vault). You may set the following environment variables:
+ * VAULT_TLS_KEY: TLS key file contents
+ * VAULT_TLS_CERT: TLS cert file contents
+ * VAULT_CONFIG_HCL: HCL config file contents
+ * VAULT_CONFIG_JSON: JSON config file contents
+
+### Step 2: Initialize the vault
+SSH into one of the DC/OS cluster nodes, and initialize the vault with the following:
+```
+$ docker run -e "VAULT_SKIP_VERIFY=true" -e "VAULT_ADDR=https://vault.marathon.mesos:8200" --entrypoint=vault -t brndnmtthws/vault-dcos init
+Key 1: 62b6e5c157446c05c067bb41fadf931fd8f422f4af2a4c0ee056acbd5a89d3ed01
+Key 2: a065dbfd663c5a619bfdc74ebce68051f9e7004d19c67bc6726daf49e209d4ea02
+Key 3: 91e97d22436508a67905e535f636bb3e4550a8ad77b9f69da43717baa4012b5b03
+Key 4: dcef79c03925db47c8494decfe959be39079ac987ac774fc5fb7148528dcb33a04
+Key 5: ed63df1f1c7c89802ab16f97b445a08c2cce047814b8f9a789edac766ed44c8b05
+Initial Root Token: 11aaf733-f280-fbaf-251d-69b9606bf4fa
+
+Vault initialized with 5 keys and a key threshold of 3. Please
+securely distribute the above keys. When the Vault is re-sealed,
+restarted, or stopped, you must provide at least 3 of these keys
+to unseal it again.
+
+Vault does not store the master key. Without at least 3 keys,
+your Vault will remain permanently sealed.
+```
+After, check to make sure it was properly initialized:
+```
+$ docker run -e "VAULT_SKIP_VERIFY=true" -e "VAULT_ADDR=https://vault.marathon.mesos:8200" --entrypoint=vault -t brndnmtthws/vault-dcos status
```

Comment:
Need to reference via virtual IP in VAULT_ADDR: <service-name>.marathon.l4lb.thisdcos.directory:<port>

Or in our case <framework-name>-vault..marathon.l4lb.thisdcos.directory:8200